app/models/course/assessment.rb

```diff
@@ -3,7 +3,10 @@
 # An assessment is a collection of questions that can be asked.
 class Course::Assessment < ActiveRecord::Base
   acts_as_lesson_plan_item
+  has_one_folder
 
+  after_initialize :build_initial_folder, if: :new_record?
+  before_validation :assign_folder_attributes
```

Comment:
Thanks !

One way I can think of is something like `has_one_folder, name_column: :title`, and let folder to figure out when to update it's name. 

But it's not flexible enough though and also only few components will use materials uploads, so I finally let the owner handle this.